File: examples/CBuilder/Compound/Locking/Server/DbGenForm.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "DbGenForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BoldHandle"
#pragma link "BoldHandles"
#pragma link "BoldPersistenceHandle"
#pragma link "BoldPersistenceHandleDB"
#pragma link "BoldSubscription"
#pragma link "BoldSystemHandle"
#pragma link "BoldAbstractDatabaseAdapter"
#pragma link "BoldAbstractPersistenceHandleDB"
#pragma link "BoldDatabaseAdapterIB"
#pragma resource "*.dfm"
TfrmDBGen *frmDBGen;
//---------------------------------------------------------------------------
__fastcall TfrmDBGen::TfrmDBGen(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmDBGen::Button1Click(TObject *Sender)
{
  BoldDatabaseAdapterIB1->CreateInterbaseDatabase();
  BoldPersistenceHandleDB1->CreateDataBaseSchema();
  ShowMessage("OK");
}
//---------------------------------------------------------------------------

File: examples/Delphi/Compound/Locking/Server/DbGenForm.pas
unit DbGenForm;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs, ModelDM,
  StdCtrls, BoldHandle, BoldPersistenceHandle,
  BoldPersistenceHandleDB, BoldSubscription,
  BoldHandles, BoldSystemHandle, DB, IBDatabase,
  BoldAbstractDatabaseAdapter, BoldDatabaseAdapterIB,
  BoldAbstractPersistenceHandleDB;

type
  TfrmDBGen = class(TForm)
    Button1: TButton;
    BoldPersistenceHandleDB1: TBoldPersistenceHandleDB;
    BoldDatabaseAdapterIB1: TBoldDatabaseAdapterIB;
    IBDatabase1: TIBDatabase;
    procedure Button1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmDBGen: TfrmDBGen;

implementation


{$R *.DFM}

procedure TfrmDBGen.Button1Click(Sender: TObject);
begin
  BoldDatabaseAdapterIB1.CreateInterbaseDatabase;
  BoldPersistenceHandleDB1.CreateDataBaseSchema;
  showmessage('OK');
end;

end.
